client/src/tutorials/ExpenseTracker/files/lib.rs

```diff
@@ -0,0 +1,107 @@
+use anchor_lang::prelude::*;
+
+// Your program Id will be added here when you enter "build" command
+declare_id!("");
+
+#[program]
+pub mod etracker {
+
+    use super::*;
+
+    pub fn initialize_expense(
+        ctx: Context<InitializeExpense>,
+        id: String,
```

Comment:
Why pass a `String` here if the program panics for values that are not parseable to `u64`?

```suggestion
        id: u64,
```

Reply:
I use string cause the id is seed as well and it's easier to process strings

Reply:
I don't think it's justified. Clients should adjust based on the on-chain program, not the other way around.

Reply:
Updated, can you check now?